ui/admin/app/controllers/scopes/scope/targets/target/add-injected-application-credential-sources.js

```diff
@@ -0,0 +1,51 @@
+import Controller from '@ember/controller';
+import { inject as service } from '@ember/service';
+import { computed } from '@ember/object';
+
+export default class ScopesScopeTargetsTargetAddInjectedApplicationCredentialSourcesController extends Controller {
+  // =services
+
+  @service intl;
+
+  // =attributes
+
+  /**
+   * Checks for unassigned credential sources.
+   * @type {boolean}
+   */
+  get hasAvailableInjectedApplicationCredentialSources() {
+    return this.filteredCredentialSources.length > 0;
+  }
+
+  /**
+   * Filter out credential sources not already added to the target.
+   * @type {[CredentialLibraryModel, CredentialModel]}
+   */
+  @computed(
+    'model.{target.injected_application_credential_source_ids.[],credentialLibraries.[],credentials.[]}'
+  )
+  get filteredCredentialSources() {
+    // Get IDs for credential sources already added to the current target
+    const currentCredentialSourceIDs =
```

Comment:
Not blocking: I personally prefer using a `Set` for these operations but this is a topic I added to the tech time to talk about. Curious what other's thoughts are.

In this case there's performance optimization here since a `filter` with `includes` can be slow _if_ there's a decent amount of values in either list, which is unlikely the case here. I still prefer set in these scenarios because it seems like we're using it like one and it's always faster.